addons/web/static/src/views/fields/badge/badge_field.js

```diff
@@ -25,6 +26,9 @@ export class BadgeField extends Component {
 
     get classFromDecoration() {
         const evalContext = this.props.record.evalContextWithVirtualIds;
+        if (this.props.colorField !== undefined) {
+            return `o_badge_color_${this.props.record.data[this.props.colorField]}`;
+        }
```

Comment:
nit picking
```suggestion
    get badgeClass() {
        if (this.props.colorField) {
            return `o_badge_color_${this.props.record.data[this.props.colorField]}`;
        }
        const evalContext = this.props.record.evalContextWithVirtualIds;
```
- classFromDecoration is no longer accurate as you do more than that now
- no need to compute the eval context if there's a colorField => move above
- the condition can be slightly simplified, no need to check !== undefined